# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Date 1701801604 -3600
# Parent  1741951c9dab340e85e60ea06b411240a35476ac
Feature: Linux system tray, special compile for FlatPak if config FLATPAK_COMPILE is set.

diff --git a/mailnews/base/src/nsMessengerUnixIntegration.cpp b/mailnews/base/src/nsMessengerUnixIntegration.cpp
--- a/mailnews/base/src/nsMessengerUnixIntegration.cpp
+++ b/mailnews/base/src/nsMessengerUnixIntegration.cpp
@@ -216,24 +216,30 @@ nsMessengerUnixIntegration::UpdateUnread
     // Work out which activation call we should use.
     nsAutoCString prefValue;
     mozilla::Preferences::GetCString("mail.minimizeToTray.desktopsGtkWindowPresentWithTime", prefValue);
     useWithTime = FindInReadable(desktop, prefValue);
 #ifdef DEBUG_RESTORE
     fprintf(stderr, "=== Desktop is %s, %susing GtkWindowPresentWithTime\n", desktop.get(), useWithTime ? "" : "not ");
 #endif
 
+#ifdef FLATPAK_COMPILE
+    // The FlatPak folks do their own icon magic.
+    strcpy(icon_path0, "eu.betterbird.Betterbird-default");
+    strcpy(icon_pathN, "eu.betterbird.Betterbird-newmail");
+#else
     // Find icon absolute path.
     size_t len = readlink("/proc/self/exe", icon_path0, PATH_MAX);
     char* q = icon_path0 + len - 1;
     while (*q != '/') q--;
     *q = 0;
     strcpy(icon_pathN, icon_path0);
     strcat(icon_path0, "/chrome/icons/default/default.svg");
     strcat(icon_pathN, "/chrome/icons/default/newmail.svg");
+#endif
 
     ci = app_indicator_new("Betterbird-systray-icon",
                            unreadCount > 0 ? icon_pathN : icon_path0,
                            APP_INDICATOR_CATEGORY_APPLICATION_STATUS);
     g_signal_connect(ci, APP_INDICATOR_SIGNAL_ACTIVATE_EVENT, G_CALLBACK(activated), NULL);
 
     // Initialise to something non-zero, subtract a second
     // so it predates any mimizing via "startup minimized".
